pkg/imgpkg/lockconfig/images_lock.go

```diff
@@ -93,3 +112,34 @@ func (c ImagesLock) WriteToPath(path string) error {
 
 	return nil
 }
+
+func (i ImageRef) Locations() []string {
+	if i.locations == nil {
+		return []string{i.Image}
+	}
+
+	locations := i.locations
```

Comment:
this actually does not copy locations. you'll have to do `locations := append([]string{}, i.locations...)`.